Source/WebKit/WebProcess/Network/WebSocketProvider.cpp

```diff
@@ -53,7 +53,9 @@ void WebSocketProvider::initializeWebTransportSession(WebCore::ScriptExecutionCo
         return completionHandler(nullptr);
     }
     ASSERT(is<Document>(context));
-    WebKit::WebTransportSession::initialize(url, WTFMove(completionHandler));
+    RefPtr<SecurityOrigin> origin = context.securityOrigin();
+    ASSERT(origin);
```

Comment:
Could you put this before the is<WorkerGlobalScope> check above?  We will need it for both cases.  Also, let's null check the origin and return completionHandler(nullptr) if it is null just in case.